packages/hoppscotch-app/pages/import.vue

```diff
@@ -0,0 +1,90 @@
+<script setup lang="ts">
+import axios from "axios"
+import * as TO from "fp-ts/TaskOption"
+import * as TE from "fp-ts/TaskEither"
+import * as RA from "fp-ts/ReadonlyArray"
+
+import { useRoute, useRouter, onMounted } from "@nuxtjs/composition-api"
+import * as E from "fp-ts/Either"
+import { pipe } from "fp-ts/function"
+import { HoppRESTRequest, HoppCollection } from "@hoppscotch/data"
+import { isType } from "~/helpers/functional/object"
+import { appendRESTCollections } from "~/newstore/collections"
+import { useToast, useI18n } from "~/helpers/utils/composables"
+import { URLImporters } from "~/helpers/import-export/import/importers"
+import { IMPORTER_INVALID_FILE_FORMAT } from "~/helpers/import-export/import"
+
+const route = useRoute()
+const router = useRouter()
+const toast = useToast()
+const t = useI18n()
+
+const IMPORTER_INVALID_TYPE = "importer_invalid_type" as const
+const IMPORTER_INVALID_FETCH = "importer_invalid_fetch" as const
+
+const importCollections = (url: unknown, type: unknown) =>
+  pipe(
+    TE.Do,
+    TE.bind("importer", () =>
+      pipe(
+        URLImporters,
+        RA.findFirst(
+          (importer) =>
+            importer.applicableTo.includes("url-import") && importer.id === type
+        ),
+        TE.fromOption(() => IMPORTER_INVALID_TYPE)
+      )
+    ),
+    TE.bindW("content", () =>
+      pipe(
+        url,
+        TO.fromPredicate(isType("string")),
+        TO.chain((url) => TO.tryCatch(() => fetchUrlData(url))),
+        TE.fromTaskOption(() => IMPORTER_INVALID_FETCH)
+      )
+    ),
+    TE.chainW(({ importer, content }) =>
+      pipe(
+        content.data,
+        TO.fromPredicate(isType("string")),
+        TE.fromTaskOption(() => IMPORTER_INVALID_FILE_FORMAT),
+        TE.chain((data) => importer.importer([data]))
+      )
+    )
+  )
+
+onMounted(async () => {
+  const { query } = route.value
+
+  const url = query.url
+  const type = query.type
+
+  const result = await importCollections(url, type)()
+
+  pipe(result, E.fold(handleImportFailure, handleImportSuccess))
+
+  router.replace("/")
+})
+
+const handleImportFailure = (error: string) => {
```

Comment:
We can see what exactly are the possible errors here (check the type of result on the onMounted block). We can do better and more specific toast messages.